File: BlogApi/Application.contract/Tags/GetTags/GetTagsRequest.cs
using System.Text.Json.Serialization;
using BlogModule.Domain.Enums;
using BlogModule.Domain.Models.Ids;
using BlogModule.Infrastructure.Binders.BlogPostId;
using Microsoft.AspNetCore.Mvc;
using Results.Contract;

namespace BlogModule.Application.Contract.Tags.GetTags;

public class GetTagsRequest : ISortable<TagSortColumnEnum?>
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public TagSortColumnEnum? OrderBy { get; init; } = TagSortColumnEnum.Name;
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public SortOrderEnum? OrderDirection { get; init; } 
    [ModelBinder(BinderType = typeof(BlogPostIdModelBinder))]
    public BlogPostId? BlogPostId { get; init; } 
}

File: BlogApi/Application.contract/Tags/GetTags/get_tags_request.go
package contract

type GetTagsRequest struct {
	SortQuery  []string `json:"sort_query" validate:"validBlogPostTagSortable"`
	BlogPostId string   `json:"blogPostId" example:"123e4567-e89b-12d3-a456-426614174000" swaggertype:"string" validate:"omitempty,len=36"`
}
